pkg/ignore/types.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ignore
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// fileIgnorePattern holds the ignorable patterns
+type fileIgnorePattern struct {
+	paths   []string
+	regExpr string
+	invert  bool
+}
+
+// Ignorable  holds the what type of ignore
+type Ignorable int
+
+const (
+	// No marks the directory or file not to be ignored
+	No Ignorable = iota
+	// Transitive marks the directory is transitive ignore so not to be skipped
+	Transitive
+	// Current marks the directory to be not having any transitive dependencies
+	// so clean be skipped from walking
+	Current
+)
+
+// FileIgnorer  helps identifying if a BundleFile needs to be ignored
+type FileIgnorer interface {
+	// CanIgnore checks file has to be ignored or not, returns true if it needs to be ignored
+	CanIgnore(path string, fi os.FileInfo) (Ignorable, error)
+}
+
+var (
+	_               FileIgnorer = (*defaultIgnorer)(nil)
+	defaultPatterns             = []string{".git", "vendor", "node_modules"}
```

Comment:
Why not do this here?
```suggestion
	defaultPatterns             = sets.NewString(".git", "vendor", "node_modules")
```

Reply:
I moved it for some reason and checking, but forgot to reset it ;)